mockup/patterns/relateditems/pattern.js

```diff
@@ -99,27 +99,30 @@ define([
       homeText: _t('home'),
       folderTypes: ['Folder'],
       selectableTypes: null, // null means everything is selectable, otherwise a list of strings to match types that are selectable
-      attributes: ['UID', 'Title', 'portal_type', 'path', 'getIcon'],
+      attributes: ['UID', 'Title', 'portal_type', 'path','getURL', 'getIcon','contenttype_class','is_folderish','review_state'],
```

Comment:
Sorry for that, i Just extrapolated from p.a.content/vocabulary.py
````
                    if key == 'path':
                        attr = 'getPath'
                    val = getattr(vocab_item, attr, None)
                    if callable(val):
                        if attr in _safe_callable_metadata:
                            val = val()
                        else:
                            continue
                    if key == 'path':
                        val = val[len(base_path):]
````

and i suggested :

````                    if key == 'thumb':
                        # geticon redefined in Plone > 5.0
                        # see https://github.com/plone/Products.CMFPlone/issues/1151
                        #     https://github.com/plone/Products.CMFPlone/issues/1074
                        has_image = getattr(vocab_item, 'getIcon', None)
                        if has_image:
                            val = os.path.join(vocab_item.getURL(),'@@images','image','tile')
````
which you did not object  so I thought that was a good way to provide the css class name from there. It is more or less just  a derived meta data value and would save us to mimic normalizeString in js (double maintainenance ?).

If you prefer the js variant, I will need your help - my js expertise  is not yet sufficient.

BTW:   val = os.path.join(vocab_item.getURL(),'@@images','image','tile') could be changed to a boolean value and be expanded in the templates accordingly (getURL+'/@@images/...'